src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1

```diff
@@ -196,6 +197,31 @@ param(
     default { throw "7-Zip signalled an unknown error (code $exitCode)" }
   }
 
-  $env:ChocolateyPackageInstallLocation = $destination
-  return $destination
+  if ((test-path variable:url) -AND (@($url).count -gt 0))
+  {
+    # If our unzip resulted in a .tar file that matches our input zip file without it's extensions, unzip it
+    # does not work with archive names that contain a '.' before the extensions
+    $matchingtarname = "$destinationNoRedirection\$((split-path -leaf $url).split('.')[0]).tar"
+    Write-Debug "Checking to see if we have a flie called $matchingtarname"
+    If ((Test-path $matchingtarname) -AND (!$RecursiveCall))

```

Comment:
In general - I wanted to keep my changes to as limited of a scope as possible:
- so that I could devise and manually run tests to ensure the reliability of the most common, known scenarios that the code was supposed to address.
- Because this code is very broadly used - allowing the new functionality to do arbitrary things like unzip all found sub-archives or have no limit on recursions seemed to me to open it up to more unexpected behaviors (and bug fixes) going forward.